### PR TITLE
Test 1336

### PR DESCRIPTION
**Pull request by sfc-gh-mkeller**
Tests #1336

### CODE REVIEW

**Comment by noamcohen97** [comment]
checkout #1349

**Comment by 3dbrows** [comment]
@sfc-gh-mkeller , #1336 is updated now and should build fine, but @noamcohen97 's #1349 does this _and_ adds Python 3.11 support.

**Comment by 3dbrows** [comment]
I've pushed a fix for the linting error to #1336 , @sfc-gh-mkeller

File: setup.py [hunk 1]
```diff
@@ -126,13 +148,15 @@ def build_extension(self, ext):
                 ext.include_dirs.append(LOGGING_SRC_DIR)
 
                 if sys.platform == "win32":
+                    if not any("/std" not in s for s in ext.extra_compile_args):
+                        ext.extra_compile_args.append("/std:c++17")
```
Comment (sfc-gh-mkeller) [inline setup.py:152]: It looks like the c++ standard has to be set on Windows now too @xhochy and @3dbrows

Comment (xhochy) [reply setup.py:152]: Yes, Arrow 10 switched to C++17 as the minimal supported C++ standard.

File: DESCRIPTION.md [hunk 2]
```diff
@@ -41,6 +45,8 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
    - When closing connection async query status checking is now parallelized
    - Fixed a bug where test logging would be enabled on Jenkins workers in non-Snowflake Jenkins machines
    - Enhanced the atomicity of write_pandas when overwrite is set to True
+  - Enhanced the atomicity of write_pandas when overwrite is set to True
+  - Bumped pyarrow dependency from >=8.0.0,<8.1.0 to >=10.0.0,<10.1.0
```
Comment (sfc-gh-aling) [inline DESCRIPTION.md:49]: those two should be moved under 2.9.1?

File: setup.py [hunk 3]
```diff
@@ -71,23 +71,45 @@ class MyBuildExt(build_ext):
         # this list should be carefully examined when pyarrow lib is
         # upgraded
         arrow_libs_to_copy = {
-            "linux": ["libarrow.so.800", "libarrow_python.so.800", "libparquet.so.800"],
+            "linux": [
+                "libarrow.so.1000",
+                "libarrow_dataset.so.1000",
```
Comment (sfc-gh-aling) [inline setup.py:76]: just curious,`*_dataset` is something new the arrow introduced?

File: src/snowflake/connector/cpp/ArrowIterator/BinaryConverter.cpp [hunk 4]
```diff
@@ -18,7 +18,7 @@ PyObject* BinaryConverter::toPyObject(int64_t rowIndex) const
 {
   if (m_array->IsValid(rowIndex))
   {
-    arrow::util::string_view sv = m_array->GetView(rowIndex);
+    std::string_view sv = m_array->GetView(rowIndex);
```
Comment (sfc-gh-aling) [inline src/snowflake/connector/cpp/ArrowIterator/BinaryConverter.cpp:21]: why we switch from `arrow::util::string_view` to `std::string_view`?

Comment (3dbrows) [reply src/snowflake/connector/cpp/ArrowIterator/BinaryConverter.cpp:21]: This seems to have been a change required by the underlying Arrow library after it upgraded to C++ 17.

Comment (noamcohen97) [reply src/snowflake/connector/cpp/ArrowIterator/BinaryConverter.cpp:21]: That's the new return value of 'GetView()' in PyArrow 10

https://arrow.apache.org/docs/cpp/api/array.html#_CPPv4NK5arrow15BaseBinaryArray7GetViewE7int64_t

**Review by sfc-gh-aling — APPROVED** [review]

**Pull request merged**